bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -381,6 +382,17 @@ public boolean start() throws IOException {
     @Override
     public List<SourceWithKeys> splitIntoBundles(
         long desiredBundleSizeBytes, PipelineOptions options) throws Exception {
+      // Update the desiredBundleSizeBytes in order to limit the
+      // number of splits to maximumNumberOfSplits.
+      long sizeEstimate = getEstimatedSizeBytes(options);
+      desiredBundleSizeBytes = Math.max(sizeEstimate / MAX_SPLIT_COUNT, desiredBundleSizeBytes);
```

Comment:
@vphani raised a concern about row key size, which can be up to 64KB each.  We'll have to note that and fix that scenario in the future.